### PR TITLE
Simple watcher in JS

### PR DESCRIPTION
**Pull request by stevencrader**
Prints console messages and appends to list

Uses dhive: https://gitlab.syncad.com/hive/dhive

### CODE REVIEW

**Comment by brianoflondon** [comment]
I'm far from a Javascript expert but in the python version you don't need to enumerate the Hive node list and if you don't, the code grabs whichever one it finds to be most reliable and switches. New api servers do sometimes come online. 

Long term I think podcastindex might want to run one.

**Comment by stevencrader** [comment]
Unfortunately the JS library doesn't have a node list. I took the list returned from python as my starting list of items.

Here is the file that the python beem library loads the node list from https://github.com/holgern/beem/blob/master/beem/nodelist.py

**Comment by brianoflondon** [comment]
I don't know this for sure, but if you do 

```
const client = new dhive.Client();
```

instead of enumerating the Nodes, I think it might just work.

File: hive-watcher-js/hive-watcher.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    console.log(message)
+
+    const list = document.getElementById("posts");
+    const item = document.createElement("li");
+    item.appendChild(document.createTextNode(message))
+    list.appendChild(item)
+}
+
+client.blockchain.getOperationsStream()
+    .on('data', (post) => handlePost(post));
```
Comment (MartinMouritzen) [inline hive-watcher-js/hive-watcher.js:44]: Nitpicking, but this can just be
.on('data', handlePost);

:)

**Comment by stevencrader** [comment]
> I don't know this for sure, but if you do
> 
> ```
> const client = new dhive.Client();
> ```
> 
> instead of enumerating the Nodes, I think it might just work.

Not for the JS library. If you don't provide an address, it tries to request it from `undefined`

image

File: hive-watcher-js/hive-watcher.js (hunk 1 above)
Comment (stevencrader) [reply hive-watcher-js/hive-watcher.js:44]: Thanks

**Comment by stevencrader** [comment]
I'll add the user filter Brian has discussed later this morning.

**Comment by brianoflondon** [comment]
The filter is to watch the followers of the podping account. I'm sure this is pretty easy to find in Javascript.

**Comment by stevencrader** [comment]
Add support for user verification.

Remove address that don't work and add address randomizer.

**Comment by brianoflondon** [comment]
Sorry to be a pain but I've changed the logic on authorisation a bit to be much simpler from a Hive points of view.

What I'm leanging toward is using only those accounts which the account ```@podping``` follows. That's easy to alter and can be done on any Hive front end. The other method of granting posting authority also has some mild security issues.

This is a curl version of what I'm doing:

```
curl --data-raw '{"id":16,"jsonrpc":"2.0","method":"call","params":["condenser_api","get_following",["podping",null,"blog",1000]]}' https://api.hive.blog/
```

**Comment by stevencrader** [comment]
What are you suggesting I change. I added the following check.

**Comment by brianoflondon** [comment]
My bad! My inability to read the completely biazare execution order of javascript led me to miss the bit at the end that does exactly what we need :-).

Actually right now I've raised an Issue with the python library because I'm getting a strange error with this api call on some of the api servers.

```
client.database.call('get_following', [validAccounts[0], null, 'blog', 10])
    .then(function (followers) {
            for (let follower of followers) {
                validAccounts = validAccounts.concat(follower.following)
            }
        }
    )
    .then(function () {
            // can pass the block number to start searching from. By default, uses the current block
            // note: using mode BlockchainMode.Latest does not seem to return data using `getOperationsStream`
            client.blockchain.getOperationsStream({mode: dhive.BlockchainMode.Irreversible})
                .on('data', handlePost);

        }
    );
```

**Pull request merged**